### DISCUSSION TITLE
while adding SKU/Item to cart, Sales representative Pop up is displaying to select

### DISCUSSION THREAD

**Discussion by TomPras** [microsoft/Dynamics365Commerce.Solutions#22] [Q&A] [answered]
We enabled the check box “Required sales representative” in functionality profile to get sales representative details. So that while scanning items to cart it will show Sales representative dialog and user will be selecting sales rep then the item will be added to cart. 

This is working as expected. but you have a new functionality built where all client teli transactions will be added to one custom view as shown below and when you select the estimate and click on “Add to sales” app bar button then this sales representative dialog pop-up for every line. This you want to suppress through code so that all lines will be added to cart without pop-up.  

 
![image](https://user-images.githubusercontent.com/120645157/207843424-75eb52d6-780e-4273-8e60-015361529880.png)

Below is the code which customer used to add lines to cart when you click on “Add to Sale” button. 
 
![image](https://user-images.githubusercontent.com/120645157/207843498-98b82814-78ff-4051-9857-aa9a92c35f5b.png)
![image](https://user-images.githubusercontent.com/120645157/207843570-d924d4d2-0a71-4cf1-9645-1dfe368dac85.png)

**Comment by madyke** [comment, marked as answer]
Hello, I reviewed this with Luke, but unfortunately it is not currently possible to suppress the "Add Sales Representative" dialog when "Required sales representative" is enabled in HQ.

If this is a required business scenario, please file an extension request for converting `GetSalesRepresentativeDialog` into a replaceable request handler.

**Comment by TomPras** [comment]
Hello Matt,

Thanks for the response,

Can we schedule a call today for a quick discussion today?

Regards,
Prashant Tomer
From: Matt Dyke ***@***.***>
Sent: 17 December 2022 05:04
To: microsoft/Dynamics365Commerce.Solutions ***@***.***>
Cc: Kumar Tomer, Prashant ***@***.***>; Author ***@***.***>
Subject: [External] Re: [microsoft/Dynamics365Commerce.Solutions] while adding SKU/Item to cart, Sales representative Pop up is displaying to select (Discussion #22)

This message is from an EXTERNAL SENDER - be CAUTIOUS, particularly with links and attachments.
________________________________

Hello, I reviewed this with Luke, but unfortunately it is not currently possible to suppress the "Add Sales Representative" dialog when "Required sales representative" is enabled in HQ.

If this is a required business scenario, please file an extension request for converting GetSalesRepresentativeDialog into a replaceable request handler.

—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_microsoft_Dynamics365Commerce.Solutions_discussions_22-23discussioncomment-2D4431333&d=DwMFaQ&c=eIGjsITfXP_y-DLLX0uEHXJvU8nOHrUK8IrwNKOtkVU&r=K31Yw0iEUi_vwd7-Myyu2llE6u0BGP9WIYbOzD8ZoF8Ndlp_kHcH5Ou43zDUxGV_&m=DG1-RgJWKap3zRSLjBcfmbXf4zSi8GuMjLmDigbZFsujM5K2HdtRcq1Rr6nSmLFP&s=dA-giItcQSENMTBOyzZJM5wHdgeY1XXhB4memK5Nre4&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_A4YOMJLP3GN2LDTFLUBL27DWNT35XANCNFSM6AAAAAAS7S6DC4&d=DwMFaQ&c=eIGjsITfXP_y-DLLX0uEHXJvU8nOHrUK8IrwNKOtkVU&r=K31Yw0iEUi_vwd7-Myyu2llE6u0BGP9WIYbOzD8ZoF8Ndlp_kHcH5Ou43zDUxGV_&m=DG1-RgJWKap3zRSLjBcfmbXf4zSi8GuMjLmDigbZFsujM5K2HdtRcq1Rr6nSmLFP&s=urYkf-Rd0Awrx1e8JfZFrEZ2bBmxl7OTParHSElLUrY&e=>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

________________________________

This message is for the designated recipient only and may contain privileged, proprietary, or otherwise confidential information. If you have received it in error, please notify the sender immediately and delete the original. Any other use of the e-mail by you is prohibited. Where allowed by local law, electronic communications with Accenture and its affiliates, including e-mail and instant messaging (including content), may be scanned by our systems for the purposes of information security and assessment of internal compliance with Accenture policy. Your privacy is important to us. Accenture uses your personal data only in compliance with data protection laws. For further information on how Accenture processes your personal data, please see our privacy statement at https://www.accenture.com/us-en/privacy-policy.
______________________________________________________________________________________

www.accenture.com

**Reply by madyke** [reply]
Hi @TomPras - Unfortunately our team does not have the bandwidth to handle all of these extension requests individually. At this point the process for requesting a new extension point is well established and the best way to make sure your request is properly tracked in our backlog. If you have further questions, please post them here on GitHub.

**Comment by TomPras** [comment]
Looping Sanjoy

Regards,
Prashant Tomer

From: Matt Dyke ***@***.***>
Sent: 17 December 2022 05:04
To: microsoft/Dynamics365Commerce.Solutions ***@***.***>
Cc: Kumar Tomer, Prashant ***@***.***>; Author ***@***.***>
Subject: [External] Re: [microsoft/Dynamics365Commerce.Solutions] while adding SKU/Item to cart, Sales representative Pop up is displaying to select (Discussion #22)

This message is from an EXTERNAL SENDER - be CAUTIOUS, particularly with links and attachments.
________________________________

Hello, I reviewed this with Luke, but unfortunately it is not currently possible to suppress the "Add Sales Representative" dialog when "Required sales representative" is enabled in HQ.

If this is a required business scenario, please file an extension request for converting GetSalesRepresentativeDialog into a replaceable request handler.

—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_microsoft_Dynamics365Commerce.Solutions_discussions_22-23discussioncomment-2D4431333&d=DwMFaQ&c=eIGjsITfXP_y-DLLX0uEHXJvU8nOHrUK8IrwNKOtkVU&r=K31Yw0iEUi_vwd7-Myyu2llE6u0BGP9WIYbOzD8ZoF8Ndlp_kHcH5Ou43zDUxGV_&m=DG1-RgJWKap3zRSLjBcfmbXf4zSi8GuMjLmDigbZFsujM5K2HdtRcq1Rr6nSmLFP&s=dA-giItcQSENMTBOyzZJM5wHdgeY1XXhB4memK5Nre4&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_A4YOMJLP3GN2LDTFLUBL27DWNT35XANCNFSM6AAAAAAS7S6DC4&d=DwMFaQ&c=eIGjsITfXP_y-DLLX0uEHXJvU8nOHrUK8IrwNKOtkVU&r=K31Yw0iEUi_vwd7-Myyu2llE6u0BGP9WIYbOzD8ZoF8Ndlp_kHcH5Ou43zDUxGV_&m=DG1-RgJWKap3zRSLjBcfmbXf4zSi8GuMjLmDigbZFsujM5K2HdtRcq1Rr6nSmLFP&s=urYkf-Rd0Awrx1e8JfZFrEZ2bBmxl7OTParHSElLUrY&e=>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

________________________________

This message is for the designated recipient only and may contain privileged, proprietary, or otherwise confidential information. If you have received it in error, please notify the sender immediately and delete the original. Any other use of the e-mail by you is prohibited. Where allowed by local law, electronic communications with Accenture and its affiliates, including e-mail and instant messaging (including content), may be scanned by our systems for the purposes of information security and assessment of internal compliance with Accenture policy. Your privacy is important to us. Accenture uses your personal data only in compliance with data protection laws. For further information on how Accenture processes your personal data, please see our privacy statement at https://www.accenture.com/us-en/privacy-policy.
______________________________________________________________________________________

www.accenture.com

**Comment by jayaprakashc079** [comment]
As product team confirmed in GitHub, unfortunately it is not currently possible to suppress the "Add Sales Representative" dialog when "Required sales representative" is enabled in HQ.

If this is a required business scenario, please file an extension request for converting GetSalesRepresentativeDialog into a replaceable request handler. You can find the steps to register extensibility request. 
https://learn.microsoft.com/en-us/dynamics365/fin-ops-core/dev-itpro/extensibility/extensibility-requests
